### PR TITLE
Update Marketplace Precompile ABI

### PR DESCRIPTION
**Pull request by JasonTulp**
This PR addresses a few small inconsistencies picked up when testing the marketplace precompile.
- Update ABI with consistent camelCase + fix compiler errors
- Change duration to None if input is 0 (Otherwise this will start and end the listing in the same block, but not clear it)
- Change marketplaceId to None if input is 0 (Otherwise the tx will fail due to no marketplace for id 0) 

This also fixes a minor bug in the marketplace pallet that allowed listings to be created with Some(0) duration, this would result in the listing never being cleared automatically

### CODE REVIEW

File: evm-precompiles/marketplace/README.md [hunk 1]
```diff
@@ -2,46 +2,48 @@
...
     function sellNftWithMarketplaceId(address collectionAddress, uint256[] calldata serialNumberIds, address buyer, address paymentAsset, uint256 fixedPrice, uint256 duration, uint32 marketplaceId) external returns (uint listingId);
-    function sellNft(address collectionAddress, uint256[] calldata serialNumberIds, address buyer, address paymentAsset, uint256 fixedPrice, uint256 duration, uint128 marketplace_id) external returns (uint listingId);
-    function sellSft(address collectionAddress, uint256[] calldata serialNumberIds, uint256[] calldata quantities, address buyer, address paymentAsset, uint256 fixedPrice, uint256 duration, uint128 marketplace_id) external returns (uint listingId);
+    function sellNft(address collectionAddress, uint256[] calldata serialNumberIds, address buyer, address paymentAsset, uint256 fixedPrice, uint256 duration, uint128 marketplaceId) external returns (uint listingId);
+    function sellSft(address collectionAddress, uint256[] calldata serialNumberIds, uint256[] calldata quantities, address buyer, address paymentAsset, uint256 fixedPrice, uint256 duration, uint32 marketplaceId) external returns (uint listingId);
```
Comment (KarishmaBothara) [inline evm-precompiles/marketplace/README.md:25]: should this be uint128? as for the rest of the places in interface we have the marketplaceId as uint128?

File: evm-precompiles/marketplace/README.md [hunk 2]
```diff
@@ -2,46 +2,48 @@
...
-    function auctionNftWithMarketplaceId(address collectionAddress, uint256[] calldata serialNumberIds, address paymentAsset, uint256 reservePrice, uint256 duration, uint256 marketplaceId);
-    function auctionNft(address collectionAddress, uint256[] calldata serialNumberIds, address paymentAsset, uint256 reservePrice, uint256 duration, uint256 marketplaceId);
-    function auctionSft(address collectionAddress, uint256[] calldata serialNumberIds, uint256[] calldata quantities, address paymentAsset, uint256 reservePrice, uint256 duration, uint256 marketplaceId);
+    function auctionNftWithMarketplaceId(address collectionAddress, uint256[] calldata serialNumberIds, address paymentAsset, uint256 reservePrice, uint256 duration, uint256 marketplaceId) external;
+    function auctionNft(address collectionAddress, uint256[] calldata serialNumberIds, address paymentAsset, uint256 reservePrice, uint256 duration, uint256 marketplaceId) external;
+    function auctionSft(address collectionAddress, uint256[] calldata serialNumberIds, uint256[] calldata quantities, address paymentAsset, uint256 reservePrice, uint256 duration, uint32 marketplaceId) external;
```
Comment (KarishmaBothara) [inline evm-precompiles/marketplace/README.md:32]: same query

**Comment by KarishmaBothara** [comment]
lgtm

**Review by KarishmaBothara — APPROVED** [review]

File: evm-precompiles/marketplace/README.md (hunk 1 above)
Comment (JasonTulp) [reply evm-precompiles/marketplace/README.md:25]: Hmm they should all be u32, (Except for events I think 128 is ok) 
will update the others too

Comment (JasonTulp) [reply evm-precompiles/marketplace/README.md]: https://github.com/futureversecom/trn-seed/pull/883/commits/d80983d037a1bb7e2c9dd2adbdb7d4a6c3f5fb7e

**Review by zees-dev — APPROVED** [review]
LGTM 👍

**Pull request merged**